lms/djangoapps/instructor_task/tasks_helper.py

```diff
@@ -779,9 +779,27 @@ def upload_grades_csv(_xmodule_instance_args, _entry_id, course_id, _task_input,
 
         if not course_grade:
             # An empty gradeset means we failed to grade a student.
-            task_progress.failed += 1
-            err_rows.append([student.id, student.username, err_msg])
-            continue
+            # TODO: create a separate track report so that instructors
+            # do not need to use grade reports to view learners' track
+            # before the course starts. We special-case for this scenario
+            # here... for now.
+            if err_msg == "User does not have access to this course" and (
+                            course.start > datetime.now().replace(tzinfo=UTC) or not course.start is None
+            ):
```

Comment:
This isn't the only caller of  `CourseGradeFactory.iter` and I wanted its functionality to remain exactly the same for other uses like the problem grade report. Of course, if we want to replicate this special case in that report, I'll move this.